crates/runtime/src/dataconnector.rs

```diff
@@ -247,61 +239,108 @@ pub async fn register_connector_factory(
 pub async fn create_new_connector(
     name: &str,
     params: HashMap<String, SecretString>,
+    secrets: Arc<RwLock<Secrets>>,
 ) -> Option<AnyErrorResult<Arc<dyn DataConnector>>> {
     let guard = DATA_CONNECTOR_FACTORY_REGISTRY.lock().await;
 
     let connector_factory = guard.get(name);
 
     match connector_factory {
-        Some(factory) => Some(factory(params).await),
+        Some(factory) => {
```

Comment:
This is the main change to filter out the prefixed parameters and autoload the secrets